orchestra/todos/api.py

```diff
@@ -25,36 +26,37 @@ def add_todolist_template(todolist_template_slug, task_id):
 
     task = Task.objects.get(id=task_id)
     template_todos = todolist_template.todos.get('items', [])
-    root_todo = Todo(
-        task=task,
-        description=todolist_template.name,
-        template=todolist_template
-    )
-    root_todo.save()
+    with transaction.atomic():
+        root_todo = Todo(
+            task=task,
+            description=todolist_template.name,
+            template=todolist_template
+        )
+        root_todo.save()
 
-    cond_props = {}
-    path = todolist_template.conditional_property_function.get(
-        'path', None)
-    if path:
-        try:
-            get_cond_props = locate(path)
-            cond_props = get_cond_props(task.project)
-        except Exception:
-            logger.exception('Invalid conditional function path.')
-    for template_todo in template_todos:
-        _add_template_todo(
-            template_todo, todolist_template, root_todo, task, cond_props)
+        cond_props = {}
+        path = todolist_template.conditional_property_function.get(
+            'path', None)
+        if path:
+            try:
+                get_cond_props = locate(path)
+                cond_props = get_cond_props(task.project)
+            except Exception:
+                logger.exception('Invalid conditional function path.')
+        for template_todo in template_todos:
+            _add_template_todo(
+                template_todo, todolist_template, root_todo, task, cond_props)
 
 
 def _to_exclude(props, conditions):
     """
     The conditions is it a list of conditions that get ORed together,
-    with properties in each dictionary getting ANDed.
+    with predicates in each dictionary getting ANDed.
     """
     any_condition_true = False
 
     for condition in conditions:
-        all_props_true = True
+        all_props_true = len(condition) > 0
```

Comment:
len(conditions) :P

Reply:
actually, this will always be True because if len(conditions) <= 0, this line won't get executed.